lib/event_store_client/store_adapter/in_memory.rb

```diff
@@ -13,16 +13,20 @@ def append_to_stream(stream_name, events, expected_version: nil) # rubocop:disab
             'eventId' => event.id,
             'data' => event.data,
             'eventType' => event.type,
-            'metadata' => event.metadata,
+            'metaData' => event.metadata,
             'positionEventNumber' => event_store[stream_name].length
           )
         end
       end
 
       def read(stream_name, direction: 'forward', start: 0, count: per_page, resolve_links: nil)
-        read_stream_forward(stream_name, start: start, count: count) if direction == 'forward'
-
-        read_stream_backward(stream_name, start: start, count: count)
+        response =
+          if direction == 'forward'
+            read_stream_forward(stream_name, start: start, count: count)
+          else
+            read_stream_backward(stream_name, start: start, count: count)
+          end
+        OpenStruct.new(body: response.to_json)
```

Comment:
If there is a possibility to change OpenStruct to Struc.
Of course its MRI but saw similar benchmarks for Ruby and Rails 
https://www.honeybadger.io/blog/how-openstruct-and-hashes-can-kill-performance/